core/files/uploads-manager.php

```diff
@@ -84,6 +84,10 @@ public function extract_and_validate_zip( $file_path, $allowed_file_types = null
 		// Returns an array of file paths.
 		$extracted = $zip_handler->extract( $file_path, $allowed_file_types );
 
+		if ( is_wp_error( $extracted ) ) {
+			return $extracted;
```

Comment:
Why at other place you throw an exception and here you return the original wp_error?

Reply:
I don't want to change the API of the function